components/eamxx/src/share/field/field_utils.hpp

```diff
@@ -119,9 +119,10 @@ void perturb (Field& f,
 // - f_out, f_in, and weight must be provided and allocated
 // - The first dimension is for the columns (COL)
 // - There can be only up to 3 dimensions of f_in
-template <typename ST>
+template <typename ST, bool AVG = 1>
 void horiz_contraction(const Field &f_out, const Field &f_in,
-                       const Field &weight, const ekat::Comm *comm = nullptr) {
+                       const Field &weight, const ekat::Comm *comm = nullptr,
+                       const Field &f_tmp = Field()) {
```

Comment:
I'm still looking through to try and understand the changes, but I think comments here on what `AVG` option is and what `f_tmp` should be would be helpful. 

Also, I don't like having a input field being called "tmp", since I associate that with some internal variable that isn't user facing, but this may be where I need to understand the impl a little more.